utils/interface_prepare.sh

```diff
@@ -125,20 +199,51 @@ check_interface(){
    fi
 }
 
-interface_ip_config(){
-   local_interface=$1
-   local_ip=$2
-   local_netmask=$3
-
-   conf_file=/etc/sysconfig/network-scripts/ifcfg-$local_interface
-   
+common_configs(){
+local_interface=$1
    cat << EOF > $conf_file
 DEVICE=$local_interface
+ONBOOT=yes
+NM_CONTROLLED=no
+EOF
+}
+
+ip_configs(){
+   local_ip=$1
+   local_netmask=$2
+
+   cat << EOF >> $conf_file
 IPADDR=$local_ip
 NETMASK=$local_netmask
 BOOTPROTO=static
-ONBOOT=yes
-NM_CONTROLLED=no
+EOF
+}
+
+bond_master_configs(){
+local_mode=$1
+local_miimon=$2
+local_updelay=$3
+local_downdelay=$4
+
+   cat << EOF >> $conf_file
+TYPE=Bond
+BONDING_OPTS="mode=$local_mode miimon=$local_miimon updelay=$local_updelay downdelay=$local_downdelay"
+DEFROUTE=yes
+IPV6INIT=no
+USERCTL=no
```

Comment:
don't you need NM_CONTROLLED=no

Reply:
i do, i apply it using the common_configs function.